mlx/backend/cuda/allocator.cpp

```diff
@@ -88,15 +90,19 @@ CudaAllocator::CudaAllocator()
           page_size,
           [](CudaBuffer* buf) { return buf->size; },
           [this](CudaBuffer* buf) { cuda_free(buf); }) {
-  // TODO: Set memory limit for multi-device.
   size_t free, total;
   CHECK_CUDA_ERROR(cudaMemGetInfo(&free, &total));
   memory_limit_ = total * 0.95;
   max_pool_size_ = memory_limit_;
+  int loc = 0;
+  CHECK_CUDA_ERROR(cudaDeviceGetDefaultMemPool(&cuda_pool_, loc));
```

Comment:
We can support multi-device with a for loop:

```c++
for (int device = device_count; device >=0; device++) {
  CHECK_CUDA_ERROR(cudaSetDevice(device));
  ...
}
```

Reply:
I don't think we need to worry about supporting multi device right now. I wouldn't actively design agains it.. but we don't have any use cases with it so I wouldn't spend too much time on it right now. For current multi GPU jobs we use multiple processes.